crates/optimism/rpc/src/eth/pending_block.rs

```diff
@@ -39,6 +46,35 @@ where
         self.inner.evm_config()
     }
 
+    /// Returns the locally built pending block
+    async fn local_pending_block(&self) -> Result<Option<SealedBlockWithSenders>, Self::Error> {
+        // See: <https://github.com/ethereum-optimism/op-geth/blob/f2e69450c6eec9c35d56af91389a1c47737206ca/miner/worker.go#L367-L375>
+        let mut lock = self.pending_block().lock().await;
+        let latest = self
+            .provider()
+            .latest_header()
+            .map_err(Self::Error::from_eth_err)?
+            .ok_or_else(|| EthApiError::UnknownBlockNumber)?;
+        let (_, block_hash) = latest.split();
+        let last_block = self
+            .provider()
+            .sealed_block_with_senders(BlockHashOrNumber::from(block_hash), Default::default())
+            .map_err(Self::Error::from_eth_err)?;
+
+        let pending_block = match last_block {
+            Some(pending_block) => pending_block,
+            None => {
+                debug!(target: "rpc", "Failed to build pending block get latest block");
+                return Ok(None)
+            }
+        };
+
+        let now = Instant::now();
+        *lock = Some(PendingBlock::new(pending_block.clone(), now + Duration::from_secs(1)));
```

Comment:
we don't need to update the lock instead this should simply return the latest block